pkg/sql/sqlbase/table.go

```diff
@@ -2596,7 +2596,13 @@ func TightenEndKey(
 
 			// We first iterate back to the previous key value
 			//    /1/#/1 --> /1/#/0
-			end = encoding.UndoPrefixEnd(end)
+			// TODO(radu): is this correct? If we have:
```

Comment:
@richardwu please comment on this, this seems fine but I'm not sure.

Reply:
This seems fine in the `ok` case.

If I understand correctly, `!ok` occurs when the last byte is `0x00` or `NULLASC`. Although logical end keys cannot end in `NULLASC` and be `!inclusive`,  in the distsql partitioning case we could eventually have a split on `/NULLASC`.

This is NYC, but we probably want to short-circuit as we did above with `NOTNULLDESC`. Regardless, we should probably return an error if it ever returns `!ok` otherwise we'd be incorrectly extending the end key to `/NULLASC/#` which includes all `NULL` values in an ascending index.

Reply:
Changed to `return end, nil` if `!ok`